### PR TITLE
Adds depth-limited search

### PR DESCRIPTION
**Pull request by jfinkels**
This pull request simply resolves the merge conflicts from #1928. It also updates the examples a bit to use keyword arguments to make them clearer.

### CODE REVIEW

File: networkx/algorithms/traversal/__init__.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from .beamsearch import *
 from .breadth_first_search import *
 from .depth_first_search import *
+from .depth_limited_search import *
```
Comment (mcognetta) [inline networkx/algorithms/traversal/__init__.py]: Why did you add this? You never added a depth_limited_search module.

Comment (jfinkels) [inline networkx/algorithms/traversal/__init__.py]: I added it by mistake when I was squashing the many previous commits while rebasing. Thanks for catching that.
